Fails to make on MacOS 10.15.7 Catalina

iMac27:~/downloads/libcyaml-main-2 % make
cc -o build/debug/libcyaml.so.1 build/debug/shared/src/mem.o build/debug/shared/src/free.o build/debug/shared/src/load.o build/debug/shared/src/save.o build/debug/shared/src/util.o build/debug/shared/src/utf8.o -L/usr/local/lib/ -lyaml  -Wl,-soname=libcyaml.so.1 -shared
ld: unknown option: -soname=libcyaml.so.1
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [build/debug/libcyaml.so.1] Error 1

--- a/.github/workflows/ci-devel.yaml
+++ b/.github/workflows/ci-devel.yaml
@@ -1,10 +1,10 @@
 name: CI-devel
 on: [push]
 jobs:
-  gcc:
+  linux-gcc:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
     - name: apt-get install packages
       run: sudo apt-get update -qq &&
            sudo apt-get install --no-install-recommends -y
@@ -21,10 +21,10 @@ jobs:
       run: make valgrind-quiet
     - name: sanitize test
       run: make VARIANT=san test
-  clang:
+  linux-clang:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
     - name: apt-get install packages
       run: sudo apt-get update -qq &&
            sudo apt-get install --no-install-recommends -y

--- a/.github/workflows/ci-devel.yaml
+++ b/.github/workflows/ci-devel.yaml
@@ -41,3 +41,25 @@ jobs:
       run: make valgrind-quiet CC=clang
     - name: sanitize test
       run: make CC=clang VARIANT=san test
+  mac-gcc:
+    runs-on: macos-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: brew install deps
+      run: brew update && brew install
+               libyaml
+    - name: build
+      run: make
+    - name: test
+      run: make test
+  mac-clang:
+    runs-on: macos-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: brew install deps
+      run: brew update && brew install
+               libyaml
+    - name: build
+      run: make CC=clang
+    - name: test
+      run: make test CC=clang

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -4,10 +4,10 @@ on:
     branches:
     - main
 jobs:
-  gcc:
+  linux-gcc:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
     - name: apt-get install packages
       run: sudo apt-get update -qq &&
            sudo apt-get install --no-install-recommends -y
@@ -24,10 +24,10 @@ jobs:
       run: make valgrind-quiet
     - name: sanitize test
       run: make VARIANT=san test
-  clang:
+  linux-clang:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
     - name: apt-get install packages
       run: sudo apt-get update -qq &&
            sudo apt-get install --no-install-recommends -y

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -44,3 +44,25 @@ jobs:
       run: make valgrind-quiet CC=clang
     - name: sanitize test
       run: make CC=clang VARIANT=san test
+  mac-gcc:
+    runs-on: macos-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: brew install deps
+      run: brew update && brew install
+               libyaml
+    - name: build
+      run: make
+    - name: test
+      run: make test
+  mac-clang:
+    runs-on: macos-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: brew install deps
+      run: brew update && brew install
+               libyaml
+    - name: build
+      run: make CC=clang
+    - name: test
+      run: make test CC=clang

--- a/Makefile
+++ b/Makefile
@@ -26,6 +26,8 @@ else
 $(error VARIANT must be 'debug' (default), 'san', or 'release')
 endif
 
+UNAME_S := $(shell uname -s)
+
 LIB_NAME = libcyaml
 LIB_PKGCON = $(LIB_NAME).pc
 LIB_STATIC = $(LIB_NAME).a
@@ -64,7 +66,7 @@ CFLAGS += -std=c11 -Wall -Wextra -pedantic \
 		-Wredundant-decls -Wundef -Wvla -Wdeclaration-after-statement
 CFLAGS += -MMD -MP
 LDFLAGS += $(LIBYAML_LIBS)
-LDFLAGS_SHARED += -Wl,-soname=$(LIB_SH_MAJ) -shared
+LDFLAGS_SHARED = -Wl,-soname=$(LIB_SH_MAJ) -shared
 
 ifeq ($(VARIANT), debug)
 	CFLAGS += -O0 -g
@@ -98,6 +100,14 @@ LIB_DEP_SHARED = $(patsubst $(BUILDDIR)%,$(BUILDDIR_SHARED)%,$(LIB_DEP))
 LIB_DEP_STATIC = $(patsubst $(BUILDDIR)%,$(BUILDDIR_STATIC)%,$(LIB_DEP))
 
 LIB_PATH = LD_LIBRARY_PATH=$(BUILDDIR)
+
+ifeq ($(UNAME_S),Darwin)
+	LIB_SHARED = $(LIB_NAME).dylib
+	LIB_SH_MAJ = $(LIB_NAME).$(VERSION_MAJOR).dylib
+	LIB_SH_VER = $(LIB_NAME).$(VERSION_STR).dylib
+	LDFLAGS_SHARED = -dynamiclib -install_name "$(LIB_SH_MAJ)" -current_version $(VERSION_STR)
+	LIB_PATH = DYLD_FALLBACK_LIBRARY_PATH=$(BUILDDIR)
+endif
 
 TEST_SRC_FILES = units/free.c units/load.c units/test.c units/util.c \
 		units/errs.c units/file.c units/save.c units/utf8.c

